src/PowerBalancerAgent.cpp

```diff
@@ -215,12 +222,21 @@ namespace geopm
         if (m_is_single_node) {
             if (m_is_first_policy) {
                 m_policy = in_policy;
+                double duration = m_policy[M_POLICY_PERIOD_DURATION];
+                m_sample_agg->period_duration(duration);
+                double warm_time = M_WARM_PERIODS_MIN * duration;
+                if (warm_time >= M_WARM_TIME_MIN) {
+                    m_warm_periods = M_WARM_PERIODS_MIN;
+                }
+                else {
+                    m_warm_periods = static_cast<int>(M_WARM_TIME_MIN / duration) + 1;
+                }
```

Comment:
The idea is that it will take about 200 periods to converge and show a benefit (at least that is how it has worked in the past), so we know that waiting 10 periods will be at most 5% of the runtime required for a benefit.